openedx/metrics-dashboard
# Issue#226 Adds Makefile and docs for running local grimoire instances

## Description

Adds the following files to make runnning a local grimoire instance easier:

* `Makefile` for commands to start, restart, stop, and destroy the `grimoirelabs/full` docker instance, which is configured to closely resemble the community production instance.
* `projects-devstack.json` minimal projects list for testing
* `credentials.cfg.sample` with base structure and docs for setting dev credentials.
* `logs/` empty dir to house SirMordred log files

Updates `README.md` with devstack setup instructions and FAQs.

## Supporting information

See https://github.com/edx/open-edx-proposals/issues/226, [FAL-2129](https://tasks.opencraft.com/browse/FAL-2129) (OpenCraft internal)

## Testing instructions

1. Follow the instructions in the updated `README.md` to [run a full local grimoire instance](https://github.com/open-craft/metrics-dashboard/blob/jill/devstack/README.md#run-full-local-instance).
2. Tail `logs/all.log` to watch the progress of ingesting data from the `projects-devstack.json` sources (this will take a little while)
3. Load a Grimoire dashboard, e.g. http://localhost:5601/app/kibana#/dashboard/Overview?_g=(), to ensure that the services are running as expected.

## Additional information

1. I don't have access to https://github.com/edx/repo-tools-data, which contains [data used to generate the `identities.yaml`](https://github.com/openedx/metrics-dashboard/blob/master/entrypoint.sh#L15) file.
   So the git authors shown in Grimoire are all in the Unknown organization, ungrouped into their respective orgs.
   
   Will resolve this issue in a follow-up task.
1. a89f2c3 uses the `sortinghat` `bots_names` config parameter to provide a list of bot names to exclude, but this parameter has been removed: https://github.com/chaoss/grimoirelab-sirmordred/issues/369

   The version of sirmordred running on the docker instance still supports this however. Will investigate how bots are being flagged in the community instance.
1. I started this task by reading the [grimoirelab-tutorial](https://chaoss.github.io/grimoirelab-tutorial/), which pointed me towards these simple `docker` containers.  However, I later found a [`docker-compose` mechanism](https://github.com/chaoss/grimoirelab/tree/master/docker-compose) which contains Hatstall , which seems closer to what is running in production? 

  See https://github.com/openedx/metrics-dashboard/pull/9 for the `docker-compose.yml` which is close to what is being used in production.
